src/librustc_mir/hair/mod.rs

```diff
@@ -213,10 +213,7 @@ pub enum ExprKind<'tcx> {
     },
     Repeat {
         value: ExprRef<'tcx>,
-        // FIXME(#29789): Add a separate hair::Constant<'tcx> so this could be more explicit about
-        // its contained data. Currently this should only contain expression of ExprKind::Literal
-        // kind.
-        count: ExprRef<'tcx>,
+        count: Constant<'tcx>,
```

Comment:
The problem with chaning it to `ConstVal` is that this leads to a problem with the MIR visitor, which does only have a `visit_constant` function. Of course we could just try to construct a `Constant` object there, but that seems not too nice, but so does the alternative, adding a new method to the visitor.